vsphere/tests/test_check.py

```diff
@@ -737,3 +746,42 @@ def test_no_infra_cache_no_perf_values(aggregator, realtime_instance, dd_run_che
         )
 
         aggregator.assert_all_metrics_covered()
+
+
+@pytest.mark.skipif(sys.version_info < (3, 0), reason="vSAN API is only available in Python 3")
+@pytest.mark.usefixtures("mock_type", "mock_threadpool", "mock_api")
+def test_vsan_metrics_included_in_check(aggregator, realtime_instance, dd_run_check):
+    realtime_instance['get_vsan'] = True
+    check = VSphereCheck('vsphere', {}, [realtime_instance])
+
+    mock_cluster = MagicMock()
+    mock_host = MagicMock()
+    mock_disk = MagicMock()
+    mock_infrastructure_cache = MagicMock()
+    check.infrastructure_cache = mock_infrastructure_cache
+    mock_infrastructure_cache.get_mors.return_value = [mock_cluster]
+    mock_cluster.name = 'hello'
+    mock_cluster.configurationEx.vsanConfigInfo.enabled = True
+    mock_cluster.host = [mock_host]
+    mock_host.name = 'world'
+    mock_host.configManager.vsanSystem.config.clusterInfo.nodeUuid = 'TestHostUUID'
+    mock_host.configManager.vsanSystem.QueryDisksForVsan.return_value = [mock_disk]
+    mock_disk.vsanUuid = 'disk'
+
+    dd_run_check(check)
+
+    aggregator.assert_metric('vsphere.vsan.cluster.time', metric_type=aggregator.GAUGE, count=1)
+    aggregator.assert_metric('vsphere.vsan.cluster.health.count', count=1)
+    aggregator.assert_metric(
+        'vsphere.vsan.cluster.example_cluster_metric', count=1, tags=['vcenter_server:FAKE', 'vsphere_cluster:hello']
```

Comment:
After these are removed from the expected metrics, can we keep this test but make sure these are not emitted?

Reply:
Adding some assertions to ensure that they are not emitted.